examples/C++/DDS/BasicConfigurationExample/HelloWorld_main.cpp

```diff
@@ -0,0 +1,212 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

Comment:
**In general**
Please, change the name of the files so they fit with the name of the directory and the example (there are already too much HelloWorldPublisher.cpp files in the repository)

Reply:
**In general**
Add much more comments, those necessary for making this Example an easy guide for a user to create its own Architecture.
Specially in headers, comment methods and variables.

Reply:
:+1: